api/server/server.go

```diff
@@ -1578,7 +1578,16 @@ func ServeApi(job *engine.Job) engine.Status {
 				chErrors <- err
 				return
 			}
-			chErrors <- srv.Serve()
+			job.Eng.OnShutdown(func() {
+				if err := srv.Close(); err != nil {
+					log.Error(err)
+				}
+			})
+			if err = srv.Serve(); err != nil && strings.Contains(err.Error(), "use of closed network connection") {
+				err = nil
+			}
+			chErrors <- err
+			srv.Close()
```

Comment:
when I looked at this the other day I was curious if we need both `srv.Close()` calls? The shutdown handler registered above will call it, and we call it directly after the return of the `srv.Serve()` call here..